v2/apiserver/internal/core/rest/events_endpoints.go

```diff
@@ -0,0 +1,215 @@
+package rest
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/brigadecore/brigade/v2/apiserver/internal/core"
+	"github.com/brigadecore/brigade/v2/apiserver/internal/lib/restmachinery"
+	"github.com/brigadecore/brigade/v2/apiserver/internal/meta"
+	"github.com/gorilla/mux"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+type EventsEndpoints struct {
+	AuthFilter        restmachinery.Filter
+	EventSchemaLoader gojsonschema.JSONLoader
+	Service           core.EventsService
+}
+
+func (e *EventsEndpoints) Register(router *mux.Router) {
+	// Create event
+	router.HandleFunc(
+		"/v2/events",
+		e.AuthFilter.Decorate(e.create),
+	).Methods(http.MethodPost)
+
+	// List events
+	router.HandleFunc(
+		"/v2/events",
+		e.AuthFilter.Decorate(e.list),
+	).Methods(http.MethodGet)
+
+	// Get event
+	router.HandleFunc(
+		"/v2/events/{id}",
+		e.AuthFilter.Decorate(e.get),
+	).Methods(http.MethodGet)
+
+	// Cancel event
+	router.HandleFunc(
+		"/v2/events/{id}/cancellation",
+		e.AuthFilter.Decorate(e.cancel),
+	).Methods(http.MethodPut)
+
+	// Cancel a collection of events
+	router.HandleFunc(
+		"/v2/events/cancellations",
+		e.AuthFilter.Decorate(e.cancelMany),
+	).Methods(http.MethodPost)
+
+	// Delete event
+	router.HandleFunc(
+		"/v2/events/{id}",
+		e.AuthFilter.Decorate(e.delete),
+	).Methods(http.MethodDelete)
+
+	// Delete a collection of events
+	router.HandleFunc(
+		"/v2/events",
+		e.AuthFilter.Decorate(e.deleteMany),
+	).Methods(http.MethodDelete)
+}
+
+func (e *EventsEndpoints) create(w http.ResponseWriter, r *http.Request) {
+	event := core.Event{}
+	restmachinery.ServeRequest(
+		restmachinery.InboundRequest{
+			W:                   w,
+			R:                   r,
+			ReqBodySchemaLoader: e.EventSchemaLoader,
```

Comment:
Where does the magic happen of asserting the request payload adheres to the corresponding schema?  (And, presumably, erroring out when validation encounters issues.)  Or, is this to come?

Reply:
`restmachinery.ServeRequest(....)` does everything, all based on the `InboundRequest` you pass to it.